src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -134,7 +213,32 @@ internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
             out int newOffset,
             uint sequenceNumber)
         {
-            throw new PlatformNotSupportedException();
+            if (offset < 0 || offset > (buffer == null ? 0 : buffer.Length))
```

Comment:
We should revert anyway since the new check is doing same thing as the count check below this if block